docs/topics/yaml-rule-structure-syntax.adoc

```diff
@@ -0,0 +1,520 @@
+
+[source,terminal]
+----
+when:
+  # when more than one tags are given, a logical AND is implied
+  hasTags: <1>
+    - "tag1"
+    - "tag2"
+----
+<1> When more than one tags is given, a logical AND is implied.
+
+===== `java` provider
+
+The `java` provider analyzes Java source code.  This provider's capabilities are `referenced` and `dependency`.
+
+.`referenced`
+
+The `referenced` capability enables the provider to find references in the source code. This capability takes two input parameters, `pattern` and `location`.
+
+[source,terminal]
+----
+when:
+  java.referenced:
+    pattern: "<pattern>" <1>
+    location: "<location>" <2>
+----
+<1> A RegEx pattern to match, for example, `org.kubernetes.*`
+<2> Specifies the exact location where the pattern needs to be matched, for example, `IMPORT`
+
+The supported locations are the following:
+
+* `CONSTRUCTOR_CALL`
+* `TYPE`
+* `INHERITANCE`
+* `METHOD_CALL`
+* `ANNOTATION`
+* `IMPLEMENTS_TYPE`
+* `ENUM_CONSTANT`
+* `RETURN_TYPE`
+* `IMPORT`
+* `VARIABLE_DECLARATION`
+
+.`dependency`
+
+The `dependency` capability  enables the provider to find dependencies for a given application. {ProductShortName} generates a list of the application's dependencies, and you can use this capability to query the list and check whether a certain dependency (within a version range) exists for the application.
+
+[source,terminal]
+----
+when:
+  java.dependency:
+    name: "<dependency_name>" <1>
+    upperbound: "<version_string>" <2>
+    lowerbound: "<version_string>" <3>
+----
+<1> Name of the dependency to search for
+<2> Upper bound on the version of the dependency
+<3> Lower bound on the version of the dependency
+
+===== `go` provider
+
+The `go` provider analyzes Golang source code.  This provider's capabilities are `referenced` and `dependency`.
+
+.`referenced`
+
+The `referenced` capability enables the provider to find references in the source code.
+
+[source,terminal]
+----
+when:
+  go.referenced: "<regex_to_find_reference>"
+----
+
+.`dependency`
+
+The `dependency` capability  enables the provider to find dependencies for an application.
+
+[source,terminal]
+----
+when:
+  go.dependency:
+    name: "<dependency_name>" <1>
+    upperbound: "<version_string>" <2>
+    lowerbound: "<version_string>" <3>
+----
+<1> Name of the dependency to search for
+<2> Upper bound on the version of the dependency
+<3> Lower bound on the version of the dependency
+
+==== Custom variables
+
+Provider conditions can have associated custom variables. You can use custom variables to capture relevant information from the matched line in the source code. The values of these variables are interpolated with data matched in the source code. These values can be used to generate detailed templated messages in a rule’s action (See _Message action_). They can be added to a rule in the `customVariables` field:
+
```

Comment:
There's no link for "Message action" and I think one would be expected. Don't think I'd use italics with a reference.